### PR TITLE
getKeyFromAddress

### PR DESCRIPTION
**Pull request by dominictarr**
In the effort of making multiserver addresses the primary way to reference servers, I have implemented a better definition of valid multiserver addresses - in the `multiserver-address` module. I now wish to have code that was processing multiserver addresses in their own adhoc way to at least use a consistent module for it. (even better, to just treat them as opaque and not parse them) so first thing is adding `getKeyFromAddress`. I also have put deprecation warnings (they only print once, to standard error) for the parse* methods.

@arj03 @regular @pietgeursen

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -72,25 +74,47 @@ var normalizeChannel = exports.normalizeChannel =
...
+      console.error('deprecated api used:'+name)
+      logged = true
+    }
+    return fn.apply(this, args)
+  }
+}
+
+var log_deprecated = false
+var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
+  if(!log_deprecated) {
```
Comment (regular) [inline index.js]: Is this a leftover from before you made the deprecate() decorator?

File: index.js [hunk 2]
```diff
@@ -72,9 +74,28 @@ var normalizeChannel = exports.normalizeChannel =
...
+    log_deprecated = true
```
Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

File: index.js [hunk 3]
```diff
@@ -72,9 +74,28 @@ var normalizeChannel = exports.normalizeChannel =
     }
   }
 
-var parseMultiServerAddress = function (data) {
+function deprecate (name, fn) {
+  var logged = false
+  return function () {
+    var args = [].slice.call(arguments)
+    if(!logged) {
+      console.error('deprecated api used:'+name)
```
Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

Comment (regular) [inline index.js]: is this a left over from before you added the deprecate() decorator?

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by regular** [comment]
github seems to be broken for me today, I can't comment on the code directly it seems ..

This looks like being a left over from before you made the deprecate() decorator:

``` js
var parseMultiServerAddress = deprecate('ssb-ref.parseMultiServerAddress', function (data) {
  if(!log_deprecated) {
    log_deprecated = true
    console.error('deprecated: parseMultiServerAddress')
  }
```

**Comment by dominictarr** [comment]
thanks @regular, fixed

**Comment by dominictarr** [comment]
merged into 2.13.0 and scuttlebot-release@13.0.0-4

**Pull request merged**